dist_run.py

```diff
@@ -219,10 +218,10 @@ def _update_padded_sequence(
     new_token: torch.Tensor,
     prompt_lengths: List[int],
 ) -> None:
-    # TODO: this is a hacky way to update the padded sequence: when there is
-    # more than one prompt, the for loop and the assignment is incompatible.
     for i in range(len(prompt_lengths)):
-        padded_sequence[i, prompt_lengths[i]] = new_token
+        prompt_lengths[i] += 1
```

Comment:
No need to increment `prompt_length` here. It was done for every rank after the step now. See below.

Reply:
removed as part of an improved solution for both pp=1 and pp=2.